src/mysql.erl

```diff
@@ -883,12 +868,11 @@ handle_query_call_reply([Rec|Recs], Query, State, ResultSetsAcc) ->
             ResultSetsAcc1 = [{Names, Rows} | ResultSetsAcc],
             handle_query_call_reply(Recs, Query, State, ResultSetsAcc1);
         #error{code = Code} when State#state.transaction_level > 0,
-                                 (Code == ?ERROR_DEADLOCK orelse
-                                  Code == ?ERROR_LOCK_WAIT_TIMEOUT) ->
+                                 Code == ?ERROR_DEADLOCK ->
             %% These errors result in an implicit rollback.
             Reply = {implicit_rollback, State#state.transaction_level,
                      error_to_reason(Rec)},
-            State2 = clear_transaction_status(State),
+            State2 = State#state{transaction_level = 1},
```

Comment:
Hey, this looks odd. The old code set transaction_level = 0 here. What's the reason for this change?

Reply:
The BEGIN is not rolled back, thus we are still in level 1. Old code assumed we exited the transaction fully and hence set the level to 0.

Reply:
Ah, right!

I am writing a test case for a transaction with deadlock and no retries (or all retries failing). This is not covered by any tests yet.